src/main/java/com/linecorp/armeria/server/PathMapping.java

```diff
@@ -126,6 +127,15 @@ static PathMapping ofCatchAll() {
     String apply(String path);
 
     /**
+     * Returns the exact path of this path mapping if it is an exact path mapping, or {@link Optional#empty}
+     * otherwise. This can be useful for services which provide logic after scanning the server's mapped
+     * services, e.g., {@link com.linecorp.armeria.server.docs.DocService}.
+     */
+    default Optional<String> exactPath() {
```

Comment:
We could perhaps mark this as `@Deprecated` because we are gonna remove it in your follow-up PR?